internal/freelist/freelist_test.go

```diff
@@ -179,6 +179,30 @@ func TestFreelist_releaseRange(t *testing.T) {
 	}
 }
 
+func TestFreeList_reload_page_dedupe(t *testing.T) {
+	buf := make([]byte, 4096)
+	f := newTestFreelist()
+	f.Init([]common.Pgid{5, 6, 8})
+
+	p := common.LoadPage(buf)
+	f.Write(p)
+
+	f2 := newTestFreelist()
+	f.Init([]common.Pgid{})
+
+	f2.Free(common.Txid(5), common.NewPage(5, common.LeafPageFlag, 0, 4))
+
+	// reload should deduplicate as a pending page when reading from p's freelist
+	f2.Reload(p)
```

Comment:
When you reload the freelist from a page, it means you already read the freelist from the page in the first place.